types/boom/boom-tests.ts

```diff
@@ -136,17 +136,53 @@ interface CustomPayload extends Boom.Payload {
 (error.output.payload as CustomPayload).custom = 'abc_123';
 
 /**
- * Test additional data
+ * Test custom data
  */
 
-// If the concrete functions do not default their generic parameter to null, too, this fails:
-const error2: Boom.BoomError = Boom.badImplementation('really bad');
+// If the concrete functions do not default their generic parameter to null, this assignment fails:
+const error2: Boom.BoomError = Boom.badImplementation('message');
```

Comment:
Can you move this above and add a test for all the concrete function initialisations without a data parameter.  It's perhaps slightly overkill but it's tidy and means everything is tested.  i.e.

    const badGatewayError = Boom.badGateway('message', {some: 'data'});
    badGatewayError.data.some;
    const badGatewayError2 = Boom.badGateway('message');

etc...